src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -763,10 +763,18 @@ def unary_expression_extractor(
                     for k, v in df_alias_dict.items():
                         if v == expr.child.name:
                             df_alias_dict[k] = updated_due_to_inheritance  # type: ignore
+            origin = self.analyze(
+                expr.child, df_aliased_col_name_to_real_col_name, parse_local_name
+            )
+            if (
+                isinstance(expr.child, (Attribute, UnresolvedAttribute))
+                and origin == quoted_name
```

Comment:
why not move this check here?
https://github.com/snowflakedb/snowpark-python/blob/266334bdcf49e701f090c8e6cf517297c20f3cfb/src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py#L364-L365